File: scalaz/src/main/scala/org/openapitools/client/api/ResponseTimeFilterFastResult.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import ResponseTimeFilterFastResult._

case class ResponseTimeFilterFastResult (
  searchId: String,
locations: List[ResponseTimeFilterFastLocation],
unreachable: List[String])

object ResponseTimeFilterFastResult {
  import DateTimeCodecs._

  implicit val ResponseTimeFilterFastResultCodecJson: CodecJson[ResponseTimeFilterFastResult] = CodecJson.derive[ResponseTimeFilterFastResult]
  implicit val ResponseTimeFilterFastResultDecoder: EntityDecoder[ResponseTimeFilterFastResult] = jsonOf[ResponseTimeFilterFastResult]
  implicit val ResponseTimeFilterFastResultEncoder: EntityEncoder[ResponseTimeFilterFastResult] = jsonEncoderOf[ResponseTimeFilterFastResult]
}

File: javascript-closure-angular/API/Client/ResponseTimeFilterFastResult.js
goog.provide('API.Client.ResponseTimeFilterFastResult');

/**
 * @record
 */
API.Client.ResponseTimeFilterFastResult = function() {}

/**
 * @type {!string}
 * @export
 */
API.Client.ResponseTimeFilterFastResult.prototype.searchId;

/**
 * @type {!Array<!API.Client.ResponseTimeFilterFastLocation>}
 * @export
 */
API.Client.ResponseTimeFilterFastResult.prototype.locations;

/**
 * @type {!Array<!string>}
 * @export
 */
API.Client.ResponseTimeFilterFastResult.prototype.unreachable;

